### ISSUE TITLE
HEVC transcode distortion with Nvidia hardware decoding 

### ISSUE THREAD

**Issue by jacobrham** [jellyfin/jellyfin#5519]
I'm having an issue with Nvidia hardware transcoding. When playing a specific title (and only this title) the stream becomes distorted about 22 minutes in.  It persists for the rest of the movie from what I can tell, based on skipping around with the seek bar Initially I suspected the card was overheating, but the issue always presents itself It's always around the same timestamp no matter if I watch to to it or seek using the seek bar, so there's something else going on.  I believe it's got something to do with hardware decoding of HEVC because disabling hardware decode for HEVC in Jellyfin makes the system transcode normally and the movie looks as one would expect. Has anyone else experienced this? I'm not totally ready to rule out that it's not an issue with my card, I picked it up used. If it were a hardware issue though I would expect other 4K HDR/HEVC content to have the same issue however, which it doesn't. Is there something I can do to both keep hardware decode enabled and display the stream correctly?

Here are some screenshots of the comparison:
With hardware HEVC Decode enabled:

![Screenshot_20210313_101848](https://user-images.githubusercontent.com/35405092/111036601-fd97bd80-83ed-11eb-9388-8e2a1a8f0f39.png)

With hardware HEVC Decode disabled:
![Screenshot_20210313_102225](https://user-images.githubusercontent.com/35405092/111036618-11432400-83ee-11eb-8f1e-265c482ea5d6.png)

**Media Info of the file**
https://pastebin.com/RX2Kxdu7

**Logs**
https://pastebin.com/atDBTqZN

**FFmpeg Logs**
https://pastebin.com/GVgLv4bi

**Stats for Nerds Screenshots**
A copy of the Playback Reporting panel:
![image](https://user-images.githubusercontent.com/35405092/111036279-7eee5080-83ec-11eb-9d77-7a963c0d1935.png)

**Server System (please complete the following information):**
 - OS: Docker on Debian
 - Jellyfin Version: 10.7.0
 - Hardware settings & device: Nvidia Quadro P400 with NVENC and tonemapping. Nvidia driver version 460.56
 - Reverse proxy: Nginx 
 - Other hardware notes: Media is stored on a btrfs filesystem, Docker containers are stored in a pseudo filesystem on top of BTFFS
**Client System (please complete the following information):**
 - PS4, desktop computer
 - OS:Manjaro Linux, Orbis OS
 - Client: Web
 - Browser Firefox 86
 - Client and Browser Version: 10.7.0

**Comment by peterspenler** [comment]
I was running into issues as well with transcoding 4k HEVC files with my GTX 960.  I managed to resolve the issues by compiling FFmpeg with the "scale_cuda" filter enabled.

To do this I had to install the "cuda" package (on ubuntu you might have to add a ppa to install it, on arch it's in the community repo). I also had to modify the `configure` script for FFmpeg with the changes in [this patch](https://aur.tuna.tsinghua.edu.cn/cgit/aur.git/tree/015-ffmpeg-cuda11.0-fix.patch?h=ffmpeg-full). Then when using configure script I used the flags `--enable-nonfree --enable-cuda-nvcc` to make sure that the "scale_cuda" filter was enabled and compiled the program.

After this I didn't have any more graphical transcoding errors.
